hgeometry/src/Data/Geometry/Polygon/Core.hs

```diff
@@ -657,3 +680,44 @@ asSimplePolygon (MultiPolygon vs _)  = vs
 numberVertices :: Polygon t p r -> Polygon t (SP Int p) r
 numberVertices = snd . bimapAccumL (\a p -> (a+1,SP a p)) (,) 0
   -- TODO: Make sure that this does not have the same issues as foldl vs foldl'
+
+
+
+--------------------------------------------------------------------------------
+-- Specialized folds
+
+-- | /O(n)/ Yield the maximum point of the polygon.
+maximum :: Ord r => SimplePolygon p r -> Point 2 r :+ p
+maximum = F.maximumBy (comparing _core) . view outerBoundaryVector
+
+-- | /O(n)/ Yield the maximum point of a polygon according to the given comparison function.
+maximumBy :: (Point 2 r :+ p -> Point 2 r :+ p -> Ordering) -> SimplePolygon p r -> Point 2 r :+ p
+maximumBy fn = F.maximumBy fn . view outerBoundaryVector
+
+-- | /O(n)/ Yield the maximum point of the polygon.
+minimum :: Ord r => SimplePolygon p r -> Point 2 r :+ p
+minimum = F.minimumBy (comparing _core) . view outerBoundaryVector
+
```

Comment:
these could just have Polygon t p r as type signature no? i.e. hole vertices can't be maximal or minimal anyway.

We should be a bit more precise what the maximum point means though? I mean this seems to produce the lexicographically largest point (max x, max y). I'm not entirely sure why that is so useful to have. Having the 'minimumBY or maximumBy functions seem more useful to me).

Reply:
(note that minimumBy/maxBy do work only for simple polygons; i.e. there we should also iterate over hole vertices) to be correct.

Reply:
Good points. I've relaxed the type signatures and written better documentation. Deleting 'minimum' and 'maximum' might be the best option, though.

Reply:
I've removed 'minimum' and 'maximum' from the export list.